backend/src/receipts.js

```diff
@@ -57,10 +93,99 @@ async function getReceiptsList(blockHash) {
+  // The receipt actions are ordered in such a way that the actions for a single receipt go
+  // one after another in a correct order, so we can collect them linearly using a moving
+  // window based on the `previousReceiptId`.
+  let receipts = [];
+  let actions;
+  let previousReceiptId = "";
+  const indexerCompatibilityTransactionActionKinds = await getIndexerCompatibilityTransactionActionKinds();
+  for (const receiptAction of receiptActions) {
+    if (previousReceiptId !== receiptAction.receipt_id) {
+      previousReceiptId = receiptAction.receipt_id;
+      actions = [];
+      const receipt = {
+        actions,
+        blockTimestamp: new BN(receiptAction.executed_in_block_timestamp)
+          .divn(10 ** 6)
+          .toNumber(),
+        gasBurnt: receiptAction.gas_burnt,
+        receiptId: receiptAction.receipt_id,
+        receiverId: receiptAction.receiver_id,
+        signerId: receiptAction.predecessor_id,
+        status: INDEXER_COMPATIBILITY_RECEIPT_ACTION_KINDS.get(
+          receiptAction.status
+        ),
+        originatedFromTransactionHash:
+          receiptAction.originated_from_transaction_hash,
+        tokensBurnt: receiptAction.tokens_burnt,
+      };
+      receipts.push(receipt);
+    }
+    actions.push({
+      args: receiptAction.args,
+      kind: indexerCompatibilityTransactionActionKinds.get(receiptAction.kind),
+    });
+  }
   return receipts;
 }
 
+async function getIncludedAndExecutedReceiptsList(blockHash) {
+  let receiptsList = [];
+  const receiptsIncludedInBlockMap = new Map();
+  const receiptsExecutedInBlockMap = new Map();
+
+  const receiptsIncludedInBlock = await generateProperReceiptsList(
+    await queryReceiptsList(blockHash)
+  );
+  const receiptsExecutedInBlock = await generateProperReceiptsList(
+    await queryExecutedReceiptsList(blockHash)
+  );
+  const receiptsIdsSet = new Set([
+    ...receiptsIncludedInBlock.map(({ receiptId }) => receiptId),
+    ...receiptsExecutedInBlock.map(({ receiptId }) => receiptId),
+  ]);
+
+  if (receiptsIdsSet.size === 0) {
+    return receiptsList;
+  }
+
+  receiptsIncludedInBlock.forEach((receipt) => {
+    receiptsIncludedInBlockMap.set(receipt.receiptId, receipt);
+  });
+  receiptsExecutedInBlock.forEach((receipt) => {
+    receiptsExecutedInBlockMap.set(receipt.receiptId, receipt);
+  });
+
+  for (receiptId of receiptsIdsSet) {
+    const includedReceipt = receiptsIncludedInBlockMap.get(receiptId);
+    const executedReceipt = receiptsExecutedInBlockMap.get(receiptId);
+
+    if (!includedReceipt && executedReceipt) {
+      receiptsList.push({
+        ...executedReceipt,
+        executionStatus: "executed",
+      });
+    } else if (includedReceipt && !executedReceipt) {
+      receiptsList.push({
+        ...includedReceipt,
+        executionStatus: "delayed",
+      });
+    } else {
+      receiptsList.push({
+        ...includedReceipt,
+        executionStatus: "success",
```

Comment:
My idea is to set execution statuses and display this receipts in one section. It you like the idea we could think about proper status names. For example if receipt included and executed in this block we can show nothing and just show tooltip with explaining.